### DISCUSSION TITLE
why`assemble!(a::Assembler, rowdofs, coldofs, Ke)` cannot work

### DISCUSSION THREAD

**Discussion by zyhsjtu** [Ferrite-FEM/Ferrite.jl#984] [General]
why`assemble!(a::Assembler, rowdofs, coldofs, Ke)` cannot work
`
   K=spzeros(12,6);
   assembler = start_assemble(K);
   Ke=[1 1 1;2 2 2; 3 3 3;4 4 4;5 5 5;6 6 6];
   assemble!(assembler, [1,2,3,4,7,8], [1,2,4], Ke);
`
with the follow error
![image](https://github.com/Ferrite-FEM/Ferrite.jl/assets/82813702/42b41ff2-0dfe-4fd2-9170-7b19be0e44d3)

**Comment by KnutAM** [comment]
Seems like assembling different rows and columns is not supported for `AssemblerSparsityPattern` which you get when calling `start_assemble(K)`, but only for the `Assembler` (which you get when calling `start_assemble(N::Int)`). The separation of rows and column dofs was implemented in #471 for `Assembler`. This should probably be possible, and at minimum a better error message should be provided. 

Thanks for reporting this!

**Reply by zyhsjtu** [reply]
Thanks for your response, but the implementation that [you mentioned](https://github.com/Ferrite-FEM/Ferrite.jl/pull/471) were alerady in the stable version [code](https://github.com/Ferrite-FEM/Ferrite.jl/blob/7868110d15acf2189fd72ae06a9e46b3cadc550d/src/assembler.jl#L46-L50), and I still don't know why my codes failed.
![image](https://github.com/Ferrite-FEM/Ferrite.jl/assets/82813702/350c0216-803b-497f-9521-2c6d97a090c2)

**Reply by KnutAM** [reply]
Yes, the following code works, which changes two things from your code
1) `start_assemble(K)` -> `start_assemble()`
2) Float64 elements in Ke (notice that I added a 6. in the last element to achieve this)

```julia
K=spzeros(12,6);  # Not used anymore
assembler = start_assemble(); 
Ke=[1 1 1;2 2 2; 3 3 3;4 4 4;5 5 5;6 6 6.]; 
assemble!(assembler, [1,2,3,4,7,8], [1,2,4], Ke);
```

**Reply by zyhsjtu** [reply]
Yes, thanks for your help，your code need the line `K=end_assemble(assembler)`  at the end of the code
`
assembler = start_assemble(); 
Ke=[1 1 1;2 2 2; 3 3 3;4 4 4;5 5 5;6 6 6.]; 
assemble!(assembler, [1,2,3,4,7,8], [1,2,4], Ke);
K=end_assemble(assembler)`
